code/datums/keybindings/emote.dm

```diff
@@ -0,0 +1,594 @@
+
+/datum/keybinding/emote/carbon
+	category = KB_CATEGORY_EMOTE_CARBON
+
+/datum/keybinding/emote/carbon/can_use(client/C, mob/M)
+	return iscarbon(M) && ..()
+
+/datum/keybinding/emote/carbon/blink
+	linked_emote = /datum/emote/living/carbon/blink
+	name = "Blink"
+
+/datum/keybinding/emote/carbon/blink_r
+	linked_emote = /datum/emote/living/carbon/blink_r
+	name = "Blink (Rapid)"
+
+/datum/keybinding/emote/carbon/clap
+	linked_emote = /datum/emote/living/carbon/clap
+	name = "Clap"
+
+/datum/keybinding/emote/carbon/cross
+	linked_emote = /datum/emote/living/carbon/cross
+	name = "Cross Arms"
+
+/datum/keybinding/emote/carbon/chuckle
+	linked_emote = /datum/emote/living/carbon/chuckle
+	name = "Chuckle"
+
+/datum/keybinding/emote/carbon/cough
+	linked_emote = /datum/emote/living/carbon/cough
+	name = "Cough"
+
+/datum/keybinding/emote/carbon/moan
+	linked_emote = /datum/emote/living/carbon/moan
+	name = "Moan"
+
+/datum/keybinding/emote/carbon/giggle
+	linked_emote = /datum/emote/living/carbon/giggle
+	name = "Giggle"
+
+/datum/keybinding/emote/carbon/gurgle
+	linked_emote = /datum/emote/living/carbon/gurgle
+	name = "Gurgle"
+
+/datum/keybinding/emote/carbon/inhale
+	linked_emote = /datum/emote/living/carbon/inhale
+	name = "Inhale"
+
+/datum/keybinding/emote/carbon/inhale/sharp
+	linked_emote = /datum/emote/living/carbon/inhale/sharp
+	name = "Inhale (Sharp)"
+
+/datum/keybinding/emote/carbon/kiss
+	linked_emote = /datum/emote/living/carbon/kiss
+	name = "Kiss" //PG13
+
+/datum/keybinding/emote/carbon/wave
+	linked_emote = /datum/emote/living/carbon/wave
+	name = "Wave"
+
+/datum/keybinding/emote/carbon/yawn
+	linked_emote = /datum/emote/living/carbon/yawn
+	name = "Yawn"
+
+/datum/keybinding/emote/carbon/exhale
+	linked_emote = /datum/emote/living/carbon/exhale
+	name = "Exhale"
+
+/datum/keybinding/emote/carbon/laugh
+	linked_emote = /datum/emote/living/carbon/laugh
+	name = "Laugh"
+
+/datum/keybinding/emote/carbon/scowl
+	linked_emote = /datum/emote/living/carbon/scowl
+	name = "Scowl"
+
+/datum/keybinding/emote/carbon/faint
+	linked_emote = /datum/emote/living/carbon/faint
+	name = "Faint"
+
+/datum/keybinding/emote/carbon/alien
+	category = KB_CATEGORY_EMOTE_ALIEN
+
+/datum/keybinding/emote/carbon/alien/can_use(client/C, mob/M)
+	return isalien(M) && ..()
+
+/datum/keybinding/emote/carbon/alien/humanoid/roar
+	linked_emote = /datum/emote/living/carbon/alien/humanoid/roar
+	name = "Roar"
+
+/datum/keybinding/emote/carbon/alien/humanoid/hiss
+	linked_emote = /datum/emote/living/carbon/alien/humanoid/hiss
+	name = "Hiss"
+
+/datum/keybinding/emote/carbon/alien/humanoid/gnarl
+	linked_emote = /datum/emote/living/carbon/alien/humanoid/gnarl
+	name = "Gnarl"
+
+/datum/keybinding/emote/carbon/brain
+	category = KB_CATEGORY_EMOTE_BRAIN
+
+/datum/keybinding/emote/carbon/brain/can_use(client/C, mob/M)
+	return isbrain(M) && ..()
+
+/datum/keybinding/emote/carbon/brain/alarm
+	linked_emote = /datum/emote/living/carbon/brain/alarm
+	name = "Alarm"
```

Comment:
Some day I will make this cleaner with ungodly amounts of macro abuse. Watch this space.